src/test/java/io/github/artsok/ReRunnerTest.java

```diff
@@ -287,26 +288,80 @@ void testParameterizedRepeaterAssertionsFailure(int value) {
         );
     }
 
+    @ProgrammaticTest
+    @ParameterizedRepeatedIfExceptionsTest(repeats = 2)
+    @ValueSource(ints = {1})
+    public void reRunTestParameterized1(int number) {
+        assertThat(number, equalTo(5));
+    }
+
+    @Test
+    void runReRunParameterized1Test() throws Exception {
+        assertTestResults("reRunTestParameterized1", 0, 1, 3, 2, 0, int.class);
+    }
+
+    @ProgrammaticTest
+    @ParameterizedRepeatedIfExceptionsTest(repeats = 2)
+    @ValueSource(ints = {1, 3, 2})
+    public void reRunTestParameterized2(int number) {
+        assertThat(number, equalTo(3));
+    }
+
+    @Test
+    void runReRunParameterized2Test() throws Exception {
+        assertTestResults("reRunTestParameterized2", 1, 2, 7, 4, 0, int.class);
+    }
+
+    @ProgrammaticTest
+    @ParameterizedRepeatedIfExceptionsTest(repeats = 30)
+    @ValueSource(ints = {1, 3, 2})
+    public void reRunTestParameterized3(int number) {
+        assertThat(number, equalTo(new Random().nextInt(500) % 3));
+    }
+
+    @Test
+    void runReRunParameterized3Test() throws NoSuchMethodException {
+        SummaryGeneratingListener testListener = runTest("reRunTestParameterized3", int.class);
+
+        assertEquals(2, testListener.getSummary().getTestsSucceededCount(), "successful test runs");
+        assertEquals(1, testListener.getSummary().getTestsFailedCount(), "failed test runs");
+        assertGreaterThan(31, testListener.getSummary().getTestsStartedCount(), "started test runs");
+        assertGreaterThan(29, testListener.getSummary().getTestsAbortedCount(), "aborted test runs");
+        assertEquals(0, testListener.getSummary().getTestsSkippedCount(), "skipped test runs");
+    }
+
+    private void assertGreaterThan(int expected, long actual, String message) {
+        assertTrue(expected < actual, String.format("expected %s to be greater than %s,%nbut was %s", message, expected, actual));
+    }
+
     private void assertTestResults(String methodName, boolean successfulTestRun, int startedTests, int abortedTests,
                                    int skippedTests) throws Exception {
+        assertTestResults(methodName,
```

Comment:
slightly more readable but first and foremost it reduces duplication with the new code for parameterized tests